### ISSUE TITLE
Installation in a step-by-step manner

### ISSUE THREAD

**Issue by SebastianPokucinski** [tdsuper/SphericalObjectDetection#2]
Hi!

I found your work great, although the process of making it work is... challenging. 

Are you planning any tutorial "how to make it work"? For typical users it takes a lot of time to understand all the installation dependencies (like included submodules), referenced models (like DCNv2) etc.

To even see a basic result I am spending fourth day learning how to use the Center-Net, because after this whole time each single library of the initial repo is already outdated and there is a lot of compatibility issues.

Best Regards,
Sebastian

**Comment by Manuelstv** [comment]
hey @Sebex0, did you manage to make it work? I'm having a lot of difficulty to run the code properly

**Comment by spokucinski** [comment]
> 

No, not at all. After two weeks I just gave up. I tried to run the whole R-CenterNet or just use the annotation processing code. Neither did it work, nor did it say why. The amount of unexpected sub-problems, missonfigurations and outdated libraries was too big. I even tried to run it in fully separated environment in Docker, but with no luck. The only thing I did was to destroy my host-machine because the amount of co-dependencies to system's libraries was so big.

The only thing I was able to successfully run was the base-line network, namely CenterNet, but it does not have that much in common with R-CenterNet.

**Comment by Manuelstv** [comment]
Thanks for the reply! That's very unfortunate! Do you know any other open-source code for object detection on spherical images?

**Comment by spokucinski** [comment]
@Manuelstv 
Precisely "spherical" - not really, that's not my point of interest. I am focused on processing of equirectangular panoramas. We can say it is a Spherical representation, I am not sure if that's what you are talking about.

If so - yeah, there are plenty of other solutions in this topic. As I am an researcher and involved in a rather academical approach - I may redirect you to the 2022 review of omnidirectional image machine learning: https://arxiv.org/abs/2205.10468 
This paper is great, fresh and extensive. If you are searching for ideas - I think it will be a perfect start :)

**Comment by xuezhongcailian** [comment]
Hello, thank you for your message and appreciation. To train this code, I would be glad to provide you with a step-by-step guide. However, could you please specify which code or model you are referring to? Additionally, if you could provide more context or details, it would be helpful for me to give you a more accurate and relevant guide to training the specific code or model you're interested in

**Comment by spokucinski** [comment]
Good Morning (at least in Poland ;))!

I would love to be more precise, but it is now over 10 months after my try-outs. Many of my problems may no longer be valid.
I think the whole thread is about "how to make it work" - it would be great if you could update the library dependencies and provide a simplified tutorial how to prepare a custom dataset and use your work with custom experiments/data. The best-case scenario for other researchers would be to get a Docker image with the environment preconfigured. The installation process of the baseline Center-Net is already complicated and adding your improvements on top of it - I did not manage to configure it properly :(

Nevertheless, I would one more time appreciate your work. Even now I am in the process of your work's continuation, but I have limited my involvement to the dataset only. What I was totally missing was the tool to annotate the images the same way you did. I switched back to the 2D-Bounding Boxes representation and test the dataset usability in some out-of-the-shelf object detectors.
